File: frontend/src/types/worklog.ts
// Соответствует enums/WorkType.java, WorkStatus.java,
// dto/worklog/WorkLogResponse.java и WorkLogRequest.java
export type WorkType =
  | "REPAIR"
  | "COMPONENT_REPLACE"
  | "CLEANING"
  | "OS_REINSTALL"
  | "MOVE"
  | "INVENTORY"
  | "WARRANTY"
  | "OTHER";

export const WORK_TYPE_LABELS: Record<WorkType, string> = {
  REPAIR: "Ремонт",
  COMPONENT_REPLACE: "Замена комплектующих",
  CLEANING: "Чистка",
  OS_REINSTALL: "Переустановка ОС",
  MOVE: "Перемещение",
  INVENTORY: "Инвентаризация",
  WARRANTY: "Гарантийное обслуживание",
  OTHER: "Другое",
};

export type WorkStatus = "OPEN" | "IN_PROGRESS" | "CLOSED";

export const WORK_STATUS_LABELS: Record<WorkStatus, string> = {
  OPEN: "Открыта",
  IN_PROGRESS: "В работе",
  CLOSED: "Закрыта",
};

export interface WorkLogResponse {
  id: number;
  equipmentId: number | null;
  equipmentInventoryNumber: string | null;
  workType: WorkType;
  description: string | null;
  status: WorkStatus;
  executorId: number | null;
  executorName: string | null;
  requestedById: number | null;
  startedAt: string | null; // ISO datetime
  finishedAt: string | null;
  timeSpentMinutes: number | null;
  usedParts: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface WorkLogRequest {
  equipmentId: number;
  workType: WorkType;
  description: string | null;
  status: WorkStatus | null;
  executorId: number | null;
  requestedById: number | null;
  startedAt: string | null;
  finishedAt: string | null;
  timeSpentMinutes: number | null;
  usedParts: string | null;
}

File: backend/src/main/java/by/GroiroTechInventory/entity/WorkLog.java
package by.GroiroTechInventory.entity;

import by.GroiroTechInventory.enums.WorkStatus;
import by.GroiroTechInventory.enums.WorkType;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "work_logs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class WorkLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "equipment_id", nullable = false)
    private Equipment equipment;

    @Enumerated(EnumType.STRING)
    @Column(name = "work_type", nullable = false, length = 40)
    private WorkType workType;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private WorkStatus status;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "executor_id")
    private User executor;

    @Column(name = "requested_by_id")
    private Long requestedById;          // пока просто ID, позже можно связать

    @Column(name = "started_at")
    private LocalDateTime startedAt;

    @Column(name = "finished_at")
    private LocalDateTime finishedAt;

    @Column(name = "time_spent_minutes")
    private Integer timeSpentMinutes;

    @Column(name = "used_parts", columnDefinition = "TEXT")
    private String usedParts;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}